src/github.com/matrix-org/dendrite/clientapi/auth/auth.go

```diff
@@ -0,0 +1,54 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/util"
+)
+
+// VerifyAccessToken verifies that an access token was supplied in the given HTTP request
+// and returns the user ID it corresponds to. Returns err if there was a fatal problem checking
+// the token. Returns resErr (an error response which can be sent to the client) if the token is invalid.
+func VerifyAccessToken(req *http.Request) (userID string, resErr *util.JSONResponse, err error) {
```

Comment:
Hmm Now that I see it in the flesh the three value return is a bit cumbersome.

Maybe just return `(userID string, errResp *util.JsonResponse)` and wrap the `error`s as 500 inside `VerifyAccessToken`?